libdevcore/RLP.cpp

```diff
@@ -149,7 +149,7 @@ unsigned RLP::length() const
 		return n - c_rlpDataImmLenStart;
 	else if (n < c_rlpListStart)
 	{
-		if ((int)m_data.size() <= n - c_rlpDataIndLenZero)
+		if ((int)m_data.size() <= n - c_rlpDataIndLenZero || m_data[1] == 0)
```

Comment:
I also was worried about such issue. 
Should not it be that a compiler would first execute this expression
"(int)m_data.size() <= n - c_rlpDataIndLenZero"  and then stops as it already true?